larray/core/session.py

```diff
@@ -262,9 +315,10 @@ def load(self, fname, names=None, engine='auto', display=False, **kwargs):
             handler = handler_cls(fname, kwargs['sep'])
         else:
             handler = handler_cls(fname)
-        objects = handler.read_items(names, display=display, **kwargs)
+        metadata, objects = handler.read_items(names, display=display, **kwargs)
```

Comment:
I load metadata from `read_items()` to not call `self._open_for_read()` and `self.close()` twice.

Reply:
Yes, I understand why you did this and it was the best move without modifying too much the FileHandler API. It's just that I think we can do better if we change it. Now, this is no big deal while that API is purely internal but I think it could be useful to make it public some day and in that case, the current API is not good enough.

Reply:
I agree